packages/mobile/App/ui/navigation/screens/patientProgramRegistration/PatientProgramRegistrationDetails.tsx

```diff
@@ -1,63 +1,149 @@
+  font-weight: 500;
+`;
 
 const DataRow = ({
   label,
   value,
 }: {
   label: TranslatedTextElement;
-  value: TranslatedTextElement | TranslatedTextElement[];
+  value: TranslatedTextElement;
 }) => {
   return (
-    <StyledView
-      margin={20}
-      marginTop={0}
-      paddingBottom={20}
-      flexDirection="row"
-      justifyContent="flex-start"
-      borderBottomWidth={1}
-      borderColor={theme.colors.BOX_OUTLINE}
-    >
-      <StyledView width="40%">
-        <StyledText fontSize={14} color={theme.colors.TEXT_MID} fontWeight={400}>
-          {label}
-        </StyledText>
-      </StyledView>
-      <StyledView width="60%">
-        {Array.isArray(value) ? (
-          value.map((x, i) => (
-            <StyledText
-              key={i}
-              width="50%"
-              marginBottom={10}
-              marginLeft={20}
-              fontSize={14}
-              color={theme.colors.TEXT_SUPER_DARK}
-              fontWeight={500}
-            >
-              {x}
-            </StyledText>
-          ))
-        ) : (
-          <StyledText
-            width="50%"
-            marginLeft={20}
-            fontSize={14}
-            color={theme.colors.TEXT_SUPER_DARK}
-            fontWeight={500}
-          >
-            {value}
-          </StyledText>
-        )}
-      </StyledView>
-    </StyledView>
+    <Row>
+      <RowLabelContainer>
+        <RowLabel>{label}</RowLabel>
+      </RowLabelContainer>
+      <RowValueContainer>
+        <RowValue>{value}</RowValue>
+      </RowValueContainer>
+    </Row>
+  );
+};
+
+const HorizontalLine = ({ marginTop = 0, marginBottom = 0 }) => (
+  <StyledView
+    borderColor={theme.colors.BOX_OUTLINE}
+    borderBottomWidth={1}
+    marginTop={marginTop}
+    marginBottom={marginBottom}
+  />
+);
+
+const PatientProgramRegistrationConditionsDetailsRow = ({ conditions }) => {
+  const { getTranslation } = useTranslation();
+
+  const initConditions = Array.isArray(conditions) ? conditions : [];
+  // We hide recorded in error conditions
+  const filteredConditions = initConditions.filter(
+    ({ conditionCategory }) =>
+      conditionCategory !== PROGRAM_REGISTRY_CONDITION_CATEGORIES.RECORDED_IN_ERROR,
+  );
+
+  // Sort alphabetically by condition name
+  const sortedConditions = sortBy(filteredConditions, ({ programRegistryCondition }) => {
+    const stringId = getReferenceDataStringId(
+      'programRegistryCondition',
+      programRegistryCondition.id,
+    );
+    return getTranslation(stringId, programRegistryCondition.name);
+  });
+
+  const groupedConditions = groupBy(sortedConditions, ({ conditionCategory }) =>
+    [
+      PROGRAM_REGISTRY_CONDITION_CATEGORIES.DISPROVEN,
+      PROGRAM_REGISTRY_CONDITION_CATEGORIES.RESOLVED,
+    ].includes(conditionCategory)
+      ? 'closed'
+      : 'open',
+  );
+  const needsDivider = groupedConditions.closed && groupedConditions.open;
+
+  const TranslatedCondition = ({ condition }) => (
```

Comment:
Any reason this can't be defined outside of the main component above so that it only gets created once?

Reply:
Ah good point, will move out